ext/js/service-worker.js

```diff
@@ -0,0 +1,141 @@
+const CHROME_ALARM_ID = "gcping_endpoint_alarm";
+const CHROME_STORAGE_ENDPOINTS_KEY = "gcping_endpoints";
+
+// when the extension is installed, add an alarm to refresh our endpoints
+chrome.runtime.onInstalled.addListener((details) => {
+  if (details.reason === chrome.runtime.OnInstalledReason.INSTALL) {
+    chrome.alarms.create(CHROME_ALARM_ID, {
+      delayInMinutes: 0,
+      periodInMinutes: 60,
```

Comment:
Without knowing much about extensions, I wonder if we could add some comments explaining what these settings mean.